tests/app/shared/issue-tables/issues-table-paginated-data.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { MatPaginator, MatPaginatorIntl } from "@angular/material";
+import { Issue } from "../../../../src/app/core/models/issue.model";
+import { TEAM_4 } from "../../../constants/data.constants";
+import {
+    ISSUE_WITH_ASSIGNEES,
+    ISSUE_WITH_EMPTY_DESCRIPTION,
+    ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY,
+    ISSUE_WITH_EMPTY_DESCRIPTION_LOW_SEVERITY
+} from "../../../constants/githubissue.constants";
+import { setAndGetPaginatedData } from '../../../../src/app/shared/issue-tables/issue-tables-paginated-data';
+
+describe('isses-table-paginated-data', () => {
+    describe('getPaginatedData()', () => {
+        const changeDetectorRefMock = {
+            detectChanges: () => null,
+            markForCheck: () => null,
+            detach: () => null,
+            reattach: () => null,
+            checkNoChanges: () => null
+        };
+        let dataSet_3: Issue[];
+        let dataSet_7: Issue[];
+        let paginator: MatPaginator;
+        const dummyTeam = TEAM_4;
+        const mediumSeverityIssueWithResponse: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION, dummyTeam);
+        const mediumSeverityIssueWithAssigneee: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_ASSIGNEES, dummyTeam);
+        const lowSeverityFeatureFlawIssue: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION_LOW_SEVERITY, dummyTeam);
+        const highSeverityDocumentationBugIssue: Issue =
+            Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY, dummyTeam);
+
+        beforeEach(() => {
+            dataSet_3 = [
+                mediumSeverityIssueWithResponse,
+                mediumSeverityIssueWithAssigneee,
+                lowSeverityFeatureFlawIssue
+            ];
+            dataSet_7 = [
+                mediumSeverityIssueWithResponse,
+                mediumSeverityIssueWithAssigneee,
+                lowSeverityFeatureFlawIssue,
+                mediumSeverityIssueWithResponse,
+                mediumSeverityIssueWithAssigneee,
+                lowSeverityFeatureFlawIssue,
+                highSeverityDocumentationBugIssue
+            ];
+            paginator = new MatPaginator(new MatPaginatorIntl(), changeDetectorRefMock);
+            paginator.pageSize = 3;
+        });
+
+        it('should set the length of paginator to the length of data', () => {
+            setAndGetPaginatedData(paginator, dataSet_3);
+            expect(paginator.length).toEqual(3);
+
+            setAndGetPaginatedData(paginator, dataSet_7);
+            expect(paginator.length).toEqual(7);
+        });
+
+        it('should return list of issues according to page index', () => {
+            paginator.pageIndex = 0;
+
+            // Returns issues index 0 to 2
+            let returnedList = setAndGetPaginatedData(paginator, dataSet_3);
+            expect(returnedList).toEqual([
```

Comment:
For the tests below, maybe it would be better to consider taking the first 3 elements of the data set instead of using **magic numbers** here! >< Something along the lines of using `splice`

Reply:
We can make use of the `paginator.pagesize` to help us here too :smile:

Reply:
I actually think the original approach is better. (i.e. better to explicitly define the expected array, rather than use `splice` and `pageSize` to compute the expected array).

The latter approach introduces somewhat non-trivial logic into the test code, and the maintainer has to check whether the test code *itself* is correct.

By explicitly defining the array, it is easier to verify the correctness of the test.

I don't think we need to be concerned about the problem of "magic constants" , because `dataset_3` and `dataset_7` are only used within this set of tests. It will not be tedious to modify constants dependent on them.

What do you think @kkangs0226 @seanlowjk ?

Reply:
I think that makes sense as well, because the logic in `setAndGetPaginatedData` itself can actually just be reduced to `splice`, which means using `splice` in our test code kind of defeats the purpose. 
As for the magic constants,  I think the name of the data sets kind of imply the length (`dataset_3` and `dataset_7`) so it is kind of clear that the number is basically the length of the dataset, so perhaps it is alright to just use `3` and `7` directly. 
@seanlowjk Let us know what you think!

Reply:
Alright then, then i think we should stick with the original approach 👍